src/com/k300/tracks/Margins.java

```diff
@@ -53,14 +37,26 @@ public boolean onTheTrack(double carX, double carY) {
         axisX = Converter.getAxisX(carX);
         axisY = Converter.getAxisY(carY);
 
-        smallDistance1 = getDistance(positiveSmallC, axisX, 0, axisY);
-        smallDistance2 = getDistance(negativeSmallC, axisX, 0, axisY);
+        smallDistance1 = getDistance(positiveSmallC, axisX, axisY);
+        smallDistance2 = getDistance(negativeSmallC, axisX, axisY);
 
-        bigDistance1 = getDistance(positiveBigC, axisX, 0, axisY);
-        bigDistance2 = getDistance(negativeBigC, axisX, 0, axisY);
+        bigDistance1 = getDistance(positiveBigC, axisX, axisY);
+        bigDistance2 = getDistance(negativeBigC, axisX, axisY);
 
         double smallDistance = smallDistance1 + smallDistance2;
         double bigDistance = bigDistance1 + bigDistance2;
         return (smallDistance > (2 * smallA) && bigDistance < (2 * bigA));
     }
+
+    private double getDistance(double x1, double x2, double y2) {
+        return Point2D.distance(x1, 0, x2, y2);
```

Comment:
Instead of 0 replace to y1

Reply:
isn't the parameter always 0

Reply:
Yeah, but I think the Distance formula need to be global